src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp

```diff
@@ -345,10 +345,13 @@ std::vector<TimerInfo> GpuQueueSubmissionProcessor::ProcessGpuDebugMarkers(
       } else {
         const GpuQueueSubmission* matching_begin_submission = FindMatchingGpuQueueSubmission(
             begin_marker_thread_id, begin_marker_post_submission_cpu_timestamp);
-        // Note that we receive submissions of a single queue in order (by CPU submission time). So
-        // if there is no matching "begin submission", the "begin" was submitted before the "end"
-        // and we lost the record of the "begin submission" (which should not happen).
-        CHECK(matching_begin_submission != nullptr);
+        // Note that we receive submissions of a single queue in order (by CPU submission time).
+        // However, if we are out of timer slot indices, we might discard submissions (if it
```

Comment:
Can we / should we discard these earlier?

Reply:
I don't know how. So I guess it is fine.

Reply:
Exactly, this is the case.